doc/source/tutorial/stats/sampling_dau.rst

```diff
@@ -0,0 +1,94 @@
+
+.. _sampling-dau:
+
+Discrete Alias Urn (DAU)
+========================
+
+.. currentmodule:: scipy.stats
+
+* Required: probability vector (PV) or the PMF along with a finite domain
+* Speed:
+
+    * Set-up: slow (linear with the vector-length)
+    * Sampling: very fast 
+
+
+DAU samples from distributions with arbitrary but finite probability vectors
+(PV) of length N. The algorithm is based on an ingenious method by A.J.
+Walker and requires a table of size (at least) N. It needs one random number
+and only one comparison for each generated random variate. The setup time for
+constructing the tables is O(N).
+
+    >>> import numpy as np
+    >>> from scipy.stats import DiscreteAliasUrn
+    >>> 
+    >>> pv = [0.18, 0.02, 0.8]
+    >>> urng = np.random.default_rng(0x8383b295ecbf0874ac910b13adcc85a0)
+    >>> rng = DiscreteAliasUrn(pv, seed=urng)
+    >>> rng.rvs()
+    2
+
+By default, the probability vector is indexed starting at 0. However, this
+can be changed by passing a ``domain`` parameter. When ``domain`` is given
+in combination with the PV, it has the effect of relocating the
+distribution from ``(0, N)`` to ``(domain[0]``, ``domain[0] + N)``.
+``domain[1]`` is ignored in this case.
+
+   >>> rng = DiscreteAliasUrn(pv, domain=(10, 13), seed=urng)
+   >>> rng.rvs()
+   12
+
+The method also works when no probability vector but a PMF is given. However
+then additionally a bounded (finite) domain must also be given.
+
+    >>> class Distribution:
+    ...     def pmf(self, x, c):
+    ...         return x**c
+    ... 
+    >>> dist = Distribution()
+    >>> rng = DiscreteAliasUrn(dist=dist, domain=(1, 10), params=(2, ),
+    ...                        seed=urng)
+    >>> rng.rvs()
+    9
+
+.. note:: As :class:`~DiscreteAliasUrn` expects PMF with signature
+          ``def pmf(x: float, ...) -> float``, it first vectorizes the
+          PMF using ``np.vectorize`` and then evaluates it over all the
+          points in the domain. But if the PMF is already vectorized,
+          it is much faster to just evaluate it at each point in the domain
+          and pass the obtained PV instead along with the domain.
+          For example, ``pmf`` methods of SciPy's discrete distributions
+          are vectorized and a PV can be obtained by doing:
+
+          >>> from scipy.stats import binom, DiscreteAliasUrn
+          >>> dist = binom(10, 0.2)  # distribution object
+          >>> domain = dist.support()  # the domain of your distribution
```

Comment:
Here we see that in the distributions we are using `support`, why do we use `domain` here?

Reply:
> Here we see that in the distributions we are using `support`, why do we use `domain` here?

You are right, `support` would be a better name if we were to be more consistent with SciPy. `domain` comes from the naming convention used by UNU.RAN. I would delay changing this until we are ready to merge because it would require a lot of cosmetic changes to docs, tests, and even benchmarks interfering a lot with the review process.

Reply:
Of course 👍

Reply:
After giving it a second thought, I remember why I named it `domain` instead of `support`: The `domain` parameter allows the user to truncate the distribution. In that case, `support` might be misleading: "should the full support be passed or any arbitrary *domain* inside the support would work?" This makes me want to use `domain` instead. Does this make sense, @tupui?

Reply:
I see, let's get another opinion on this. On my side I would be fine with `support` as in the end it is the new support of the distribution.

Reply:
> After giving it a second thought, I remember why I named it domain instead of support: The domain parameter allows the user to truncate the distribution. In that case, support might be misleading: "should the full support be passed or any arbitrary domain inside the support would work?" This makes me want to use domain instead. Does this make sense,

I had the same thought and would be in favor of keeping `domain`